src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java

```diff
@@ -105,6 +105,9 @@ public void all_classes_should_be_referenced() {
             // doesn't see accesses to static fields
             "com.virtuslab.gitmachete.frontend.defs")
         .and().doNotBelongToAnyOf(classesReferencedFromPluginXmlAttributes)
+        .and()
+        .doNotBelongToAnyOf(com.virtuslab.gitmachete.frontend.actions.traverse.TraverseSyncToParent.class,
```

Comment:
Hmmm are these classes truly unreferenced? or a false positive in the test?

Reply:
tested it again after the refactor from static to regular classes - still needed, and still a false positive imho

Reply:
I think I know how to fix it... gimme a second...

Reply:
Okay just apply the following:

```diff
diff --git a/src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java b/src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java
index 053f1d08..216769d4 100644
--- a/src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java
+++ b/src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java
@@ -1,5 +1,6 @@
 package com.virtuslab.archunit;

+import static com.tngtech.archunit.core.domain.JavaModifier.SYNTHETIC;
 import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
 import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;

@@ -98,16 +99,14 @@ public class ClassStructureTestSuite extends BaseArchUnitTestSuite {
   public void all_classes_should_be_referenced() {
     val classesReferencedFromPluginXmlAttributes = extractAllClassesReferencedFromPluginXmlAttributes().toArray(Class[]::new);
     classes()
-        .that().resideOutsideOfPackages(
+        .that().doNotHaveModifier(SYNTHETIC)
+        .and().resideOutsideOfPackages(
             // Classes in *.impl.* packages may be instantiated via RuntimeBinding
             "..impl..",
             // For some reason, ArchUnit (com.tngtech.archunit.core.domain.JavaClass.getAccessesFromSelf)
             // doesn't see accesses to static fields
             "com.virtuslab.gitmachete.frontend.defs")
         .and().doNotBelongToAnyOf(classesReferencedFromPluginXmlAttributes)
-        .and()
-        .doNotBelongToAnyOf(com.virtuslab.gitmachete.frontend.actions.traverse.TraverseSyncToParent.class,
-            com.virtuslab.gitmachete.frontend.actions.traverse.TraverseSyncToRemote.class)
         // SubtypingBottom is processed by CheckerFramework based on its annotations
         .and().doNotHaveFullyQualifiedName(com.virtuslab.qual.internal.SubtypingBottom.class.getName())
         .should(new BeReferencedFromOutsideItself())
```

Reply:
ok